src/api/api_manager_impl.cpp

```diff
@@ -154,14 +155,14 @@ utils::Status ApiManagerImpl::process(const std::string &query,
 
     // Image processing phase 3 (adjustments, effects, etc.)
     image = image | embed | rotation | brightness | contrast | gamma | sharpen |
-            filter | blur | tint | background | mask;
+            filter | blur | tint | background | mask | saturate;
```

Comment:
Saturation should not be done at the last stage. It should probably be done between the `brightness` and `tint` operation.